keps/sig-node/3085-pod-conditions-for-starting-completition-of-sandbox-creation/README.md

```diff
@@ -1,1618 +1,1618 @@
+ of a pod (similar to Scheduled, Ready, etc), pod conditions is the ideal place
+ to surface these and aligns well with the existing conditions like
+ `ContainersReady` and overall `Ready`.
+ 
+ A dedicated annotation on the pod for surfacing this data is another potential
+ approach. However, usage of annotations for Kubelet managed data is typically
+ discouraged.
+ 
+ ### Surface pod sandbox creation latency instead of timestamps
+ 
+ Surfacing the amount of time it took to successfully create a pod sandbox is an
+ alternative to surfacing the condition around completion of pod
+ sandbox creation (whose delta from pod scheduled condition reflects the
+ latency). The latency data would surface the same information from a pod
+ initialization SLI perspective as mentioned in the Motivations section.
+ Implementing this approach would require an API change on the pod status to
+ surface the latency data (as this no longer fits the structure of a pod
+ condition). This data cannot be consumed by other controllers as mentioned in
+ User Stories section.
+ 
+ ### Report sandbox creation latency as an aggregated metric
+ 
+ The duration it took pod sandbox to become ready can be directly reported as a
+ prometheus metrics aggregated in a histogram. However, aggregating the data at
+ the Kubelet level prevents a metric collection service from classifying the data
+ based on interesting fields on a pod (runtime class, storage class of PVCs,
+ number of PVCs, etc) or using custom labels and annotations on pods that
+ indicate workload characteristics (that the cluster operator may wish to use as
+ a basis for aggregating the metrics).
+ 
+ This also prevents other controllers from acting on sandbox status as
+ mentioned in User Stories section.
+ 
+ ### Report sandbox creation stages using Kubelet tracing
+ 
+ The Kubelet is being instrumented to emit traces based on OpenTelemetry around
+ sandbox creation stages (as well several other parts of the pod lifecycle).
+ 
+ To implement the pod sandbox creation latency SLI/SLO use cases, the tracing
+ infrastructure needs to be able to:
+ - Collect all traces around CRI sandbox creation for all pods with no sampling.
+ - Look-up pod fields from API server (associated with a pod's trace) like
+ labels/annotations/storage classes of PVCs referred by the pod/runtimeclass/etc.
+ that is of interest to cluster operators and their users for classifying and
+ aggregating the metrics.
+ - Look-up a pod's Scheduled condition fields to determine the beginning of pod
+ sandbox creation.
+ 
+ Since the lookup of the pod fields and existing conditions is necessary for SLIs
+ around pod sandbox creation latency, surfacing the `PodHasNetwork` condition in
+ pod status will allow a metric collection service to directly access the
+ relevant data without requiring the ability to collect and parse OpenTelemetry
+ traces. As mentioned in the User Stories, popular community managed services
```

Comment:
But if the user is looking for more granular data on timings, surely a common tracing standard is the answer?

How granular would we eventually go with conditions vs tracing?